File: FrontEnd/lib/domain/interfaces/usecases/weather/i_get_weather_forecast_use_case.dart
/// @file i_get_weather_forecast_use_case.dart
/// @description Interfaz del caso de uso para obtener la previsión meteorológica.
/// @module Weather
/// @layer Domain
library;

import '../../../../domain/entities/weather_data.dart';

abstract interface class IGetWeatherForecastUseCase {
  /// Devuelve la previsión de [days] días para [location].
  ///
  /// [returns] [WeatherData] con el campo forecast poblado.
  /// [throws]  AppError.network si no hay conexión y no hay caché válido.
  Future<WeatherData> execute(String location, {int days = 7});
}

File: BackEnd/src/domain/interfaces/usecases/weather/IGetWeatherForecastUseCase.ts
/**
 * @file IGetWeatherForecastUseCase.ts
 * @description Interfaz del caso de uso para obtener la previsión meteorológica por horas.
 * @module Weather
 * @layer Domain
 */

import type { WeatherQueryRequestDTO } from '../../../dtos/weather/weather-query-request.dto.js';
import type { ForecastResponseDTO } from '../../../dtos/weather/forecast-response.dto.js';

export interface IGetWeatherForecastUseCase {
  /**
   * @param dto — Coordenadas y horas de previsión solicitadas.
   * @returns ForecastResponseDTO con la lista de condiciones horarias.
   * @throws {ValidationException} Si las coordenadas o el rango de horas son inválidos.
   * @throws {ExternalServiceException} Si la API no está disponible.
   */
  execute(dto: WeatherQueryRequestDTO): Promise<ForecastResponseDTO>;
}
